n2y/notion.py

```diff
@@ -160,14 +162,45 @@ def get_class_list(self, notion_object, object_type=None):
         except KeyError:
             raise NotImplementedError(f'Unknown "{notion_object}" class of type "{object_type}"')
 
-    def instantiate_class(self, notion_object, object_type, *args, **kwargs):
+    def instantiate_class(self, *args, **kwargs):
+        first_arg = args[0]
+        if type(first_arg) == str:
```

Comment:
If we want to use `instantiate_class` for both pandoc and notion, then we should probably make it clearer what `first_arg` is. Perhaps even explicitly pull it out of `*args` like:

```
def instantiate_class(self, payload, *args, **kwargs):
    instantiating_notion = payload == str
    if instantiating_notion:
        ...
```
or even more explicit, add a new argument.

```
def instantiate_class(self, instantiating_notion, *args, **kwargs):
    if instantiating_notion:
        ...
```

Reply:
BTW, I'm not opposed to just having two separate methods `instantiate_class_from_notion` and `instantiate_class_from_pandoc` and not having `instantiate_class` at all. Or perhaps leaving `instantiate_class` as is and just adding `instantiate_class_from_pandoc`. 

I think I could be convinced either way but this seems like another approach worth considering.

Reply:
I like the idea of explicitly pulling the first argument from args. I've assigned it the name `class_to_instantiate`, as the argument will either literally be a pandoc class to instantiate or a string denoting which class to instantiate. Instantiate class now looks like this:
```
def instantiate_class(self, class_to_instantiate, *args, **kwargs):
        if type(class_to_instantiate) == str:
            return self._instantiate_class_from_notion(class_to_instantiate, args[0], *args[1:], **kwargs)
        else:
            return self._instantiate_class_from_pandoc(class_to_instantiate, args[0], **kwargs)
```